main.go

```diff
@@ -39,20 +40,44 @@ import (
 	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
 )
 
+type ctxKey string
+
 var (
 	version    string = "snapshot"
 	debug             = kingpin.Flag("debug", "Enable debug logging").Bool()
 	local             = kingpin.Flag("local", "Run locally for development").Bool()
 	kubeconfig        = kingpin.Flag("kubeconfig", "Path to kubeconfig").OverrideDefaultFromEnvar("KUBECONFIG").String()
+	ns                = ctxKey("namespace")
+	pvcname           = ctxKey("volumeClaimName")
+	volname           = ctxKey("volumeName")
 )
 
 func init() {
 	kingpin.Version(version)
 	kingpin.Parse()
 }
 
+func logger(ctx context.Context, err error) *log.Entry {
```

Comment:
Since you're returning an `log.Entry` anyways, just accept the context and chain a `WithError()` if an error exists.

I would also rename this function to `WithCtx()`. 

So a line with an error would look like:
`WithCtx(ctx).WithError(err).Error("failed to do a thing")

Reply:
And a line without an error would look like:
`WithCtx(ctx).Info("doing thing")`

Reply:
Much better, thanks!